demos/projects/ESPRESSIF/aziotkit/main/azure_iot_freertos_esp32_main.c

```diff
@@ -0,0 +1,390 @@
+    {
+        pxNetif = esp_netif_next( pxNetif );
+        if ( prvIsOurNetif( TAG, pxNetif ) )
+        {
+            ESP_LOGI( TAG, "Connected to %s", esp_netif_get_desc( pxNetif ) );
+
+            ESP_ERROR_CHECK( esp_netif_get_ip_info( pxNetif, &xIpInfo ) );
+
+            ESP_LOGI( TAG, "- IPv4 address: " IPSTR, IP2STR( &xIpInfo.ip ) );
+        }
+    }
+
+    return ESP_OK;
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Callback to confirm time update through NTP.
+ */
+static void prvTimeSyncNotificationCallback( struct timeval * pxTimeVal )
+{
+    ( void ) pxTimeVal;
+    ESP_LOGI( TAG, "Notification of a time synchronization event" );
+    xTimeInitialized = true;
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Updates the device time using NTP.
+ */
+static void prvInitializeTime()
+{
+    sntp_setoperatingmode( SNTP_OPMODE_POLL );
+    sntp_setservername( 0, SNTP_SERVER_FQDN );
+    sntp_set_time_sync_notification_cb( prvTimeSyncNotificationCallback );
+    sntp_init( );
+
+    ESP_LOGI( TAG, "Waiting for time synchronization with SNTP server" );
+
+    while ( !xTimeInitialized )
+    {
+        vTaskDelay( pdMS_TO_TICKS( 1000 ) );
+    }
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Property message callback handler
+ */
+void vHandleProperties( AzureIoTHubClientPropertiesResponse_t * pxMessage,
+                        void * pvContext )
+{
+    ( void ) pvContext;
+
+    ESP_LOGI( TAG, "Property document payload : %.*s \r\n",
+              pxMessage->ulPayloadLength,
+              ( const char * ) pxMessage->pvMessagePayload );
+
+    switch( pxMessage->xMessageType )
+    {
+        case eAzureIoTHubPropertiesRequestedMessage:
+            ESP_LOGI( TAG, "Device property document GET received\r\n" );
+
+            break;
+        case eAzureIoTHubPropertiesWritablePropertyMessage:
+            ESP_LOGI( TAG, "Device writeable property received\r\n" );
+
+            vSampleHandleWritablePropertiesUpdate( pxMessage );
+
+            break;
+
+        case eAzureIoTHubPropertiesReportedResponseMessage:
+            ESP_LOGI( TAG, "Device reported property response received\r\n" );
+            break;
+
+        default:
+            ESP_LOGE( TAG, "Unknown property message: 0x%08x\r\n", pxMessage->xMessageType );
+            configASSERT( false );
+    }
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Implements the sample interface for generating reported properties payload.
+ */
+uint32_t ulCreateReportedPropertiesUpdate( uint8_t * pucPropertiesData,
+                                           uint32_t ulPropertiesDataSize )
+{
+    uint32_t lBytesWritten = 0;
+
+    if ( xUpdateDeviceProperties )
+    {
+        lBytesWritten = lGenerateDeviceInfo( pucPropertiesData, ulPropertiesDataSize );
+

```

Comment:
The telemetry frequency needs to be reported at startup, but only once the full device twin is processed. Would that be here?

Reply:
Telemetry is working fine now. Thanks Ryan!